applications/tari_app_utilities/src/identity_management.rs

```diff
@@ -0,0 +1,189 @@
+// Copyright 2020. The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use crate::utilities::ExitCodes;
+use log::*;
+use rand::rngs::OsRng;
+use std::{
+    clone::Clone,
+    fs,
+    path::{Path, PathBuf},
+    string::ToString,
+    sync::Arc,
+};
+use tari_comms::{multiaddr::Multiaddr, peer_manager::PeerFeatures, NodeIdentity};
+use tari_core::transactions::types::PrivateKey;
+use tari_crypto::{
+    keys::SecretKey as SecretKeyTrait,
+    tari_utilities::{hex::Hex, message_format::MessageFormat},
+};
+
+pub const LOG_TARGET: &str = "tari_application";
+
+/// Loads the node identity, or creates a new one if the --create-id flag was specified
+/// ## Parameters
+/// `identity_file` - Reference to file path
+/// `public_address` - Network address of the base node
+/// `create_id` - Whether an identity needs to be created or not
+/// `peer_features` - Enables features of the base node
+///
+/// # Return
+/// A NodeIdentity wrapped in an atomic reference counter on success, the exit code indicating the reason on failure
+pub fn setup_node_identity(
+    identity_file: &PathBuf,
```

Comment:
nit: While we're moving utility functions around I'd recommend making them generic on `AsRef<Path>` so that you can write e.g. `setup_node_identity("/tmp/file.tmp", ...)`. Same for `load_identity` - just for the sake of ease of use and consistency

```suggestion
pub fn setup_node_identity<P: AsRef<Path>>(
    identity_file: P,
```